### PR TITLE
Improve Repository handling of spaces

### PR DESCRIPTION
**Pull request by muellerzr**
Closes https://github.com/huggingface/huggingface_hub/issues/672

<s>Only thing I might need some help with is staging the tests, as this was a silent failure. Right now it err's out with `InvalidToken`. Is there something different I should be doing? TIA</s>

### CODE REVIEW

**Comment by muellerzr** [comment]
Sorry for the noise, un-drafting. Auto-import was wrong and made all the tests fail. 

Let me know if we *shouldn't* warn the user that we're creating a new repo

File: src/huggingface_hub/repository.py [hunk 1]
```diff
@@ -543,14 +556,21 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
...
+                        )
+                        api.create_repo(
+                            repo_id,
+                            token=token,
+                            repo_type=self.repo_type,
+                            organization=namespace,
+                            exist_ok=True,
+                            private=self.private,
+                            space_sdk=space_sdk,
+                        )
```
Comment (LysandreJik) [inline src/huggingface_hub/repository.py:573]: Not super enthusiastic about that part with the double API call but I understand it's needed for spaces we want to clone when no `space_sdk` is passed.

File: src/huggingface_hub/repository.py [hunk 2]
```diff
@@ -543,14 +556,21 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
...
+                    except HTTPError:
+                        logger.warning(
+                            f"Repo {namespace}/{repo_id} does not exist, creating a new repo"
+                        )
+                        api.create_repo(
+                            repo_id,
+                            token=token,
+                            repo_type=self.repo_type,
+                            organization=namespace,
+                            exist_ok=True,
```
Comment (LysandreJik) [inline src/huggingface_hub/repository.py:570]: I guess this can be removed now

**Review by LysandreJik — APPROVED** [review]
Thanks for working on it!

**Comment by LysandreJik** [comment]
Closing for same reason as https://github.com/huggingface/huggingface_hub/pull/852

**Pull request closed without merging**